### PR TITLE
Added in the Mapper dateFromFormat and convertDate methods. Added uni…

### PR DESCRIPTION
**Pull request by BertrandF23**
…t tests

### CODE REVIEW

File: Tests/Functional/Tools/Mapper/MapperTest.php [hunk 1]
```diff
@@ -123,6 +123,79 @@ public function dataProviderForCorrectMappings()
...
+                ],
+                'mapping_name' => 'tests',
+                'mapped_values' => new SerializableArray([]),
+                'context' => [],
+                'expected_value' => [
+                    'test_0' => \DateTime::createFromFormat(\DateTime::ISO8601, '2018-03-23T00:00:00+00:00'),
+                    'test_1' => \DateTime::createFromFormat(\DateTime::ISO8601, '2018-03-23T20:15:30+00:00'),
+                ],
+            ],
+            'Test dateFromFormat' => [
```
[the remark this answers is not in the document]
Comment (melmccann) [reply Tests/Functional/Tools/Mapper/MapperTest.php]: Thanks Shane... I seem to have missed a few things. I will talk to Bertrand about these.

[the remark this answers is not in the document]
Comment (BertrandF23) [reply Tests/Functional/Tools/Mapper/MapperTest.php]: The block has been duplicated. Will remove the useless one.

File: Tools/Mapper/Mapper.php [hunk 2]
```diff
@@ -194,6 +194,45 @@ public function timestampToDate($timestamp)
...
+     * Ex: dateFromFormat('d/m/Y', '23/03/2018').
+     *
+     * @param $format
+     * @param $date
+     *
+     * @return bool|\DateTime
+     */
+    public function dateFromFormat($format, $date)
+    {
+        if (!empty($format) && !empty($date)) {
```
[the remark this answers is not in the document]
Comment (BertrandF23) [reply Tools/Mapper/Mapper.php]: David did like this for stringToDate in 2016.

[the remark this answers is not in the document]
Comment (BertrandF23) [reply Tools/Mapper/Mapper.php]: I can't find who was supposed to do this review: Mel or Shane?

File: Tools/Mapper/Mapper.php [hunk 3]
```diff
@@ -194,6 +194,45 @@ public function timestampToDate($timestamp)
...
+            return \DateTime::createFromFormat('!'.$format, $date);
+        }
+    }
+
+    /**
+     * Converts a date from a format to another.
+     * Ex: convertDate('y/m/d', 'Y-m-d', '25/12/0000').
```
[the remark this answers is not in the document]
Comment (BertrandF23) [reply Tools/Mapper/Mapper.php]: Stupid example anyway, Jesus was not born at Christmas, and not exactly in year 0 nor 1. Will change to next Christmas.

File: Tools/Mapper/Mapper.php [hunk 4]
```diff
@@ -194,6 +194,45 @@ public function timestampToDate($timestamp)
...
+     *
+     * @param string $inputFormat
+     * @param string $outputFormat
+     * @param string $date
+     *
+     * @return null|string
+     */
+    public function convertDate($inputFormat, $outputFormat, $date)
+    {
```
[the remark this answers is not in the document]
Comment (BertrandF23) [reply Tools/Mapper/Mapper.php]: I like it, it's clearer and faster. And with unit testing, we can rely on it.

**Comment by BertrandF23** [comment]
These late changes will be added in next Pull Request (feature/throttle-delay-2).

File: Tools/Mapper/Mapper.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (melmccann) [reply Tools/Mapper/Mapper.php]: I think Shane had made comments Thursday but they didn't show up so I took it on Friday. Anyway, we can take it as a new review for the changes. Thanks.

**Pull request merged**